operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+			InitContainers: &redpandav1alpha2.InitContainers{
+				SetDataDirOwnership: &redpandav1alpha2.SetDataDirOwnership{
+					Enabled:           ptr.To(src.Spec.NodePoolSpec.BrokerTemplate.SetDataDirectoryOwnership),
+					Resources:         nil,
+					ExtraVolumeMounts: nil,
+				},
+				FsValidator: &redpandav1alpha2.FsValidator{
+					Enabled:           ptr.To(src.Spec.NodePoolSpec.BrokerTemplate.ValidateFilesystem),
+					Resources:         nil,
+					ExtraVolumeMounts: nil,
+					ExpectedFS:        nil,
+				},
+				Configurator:                      nil,
+				ExtraInitContainers:               nil,
+				SetTieredStorageCacheDirOwnership: nil,
+				Tuning:                            nil,
+			},
+			Replicas:    src.Spec.NodePoolSpec.Replicas,
+			PodTemplate: convertTo[*redpandav1alpha2.PodTemplate](src).(*redpandav1alpha2.PodTemplate),
+
+			AdditionalRedpandaCmdFlags:    nil,
+			AdditionalSelectorsLabels:     nil,
+			Annotations:                   nil,
+			Budget:                        nil,
+			ExtraVolumeMounts:             nil,
+			ExtraVolumes:                  nil,
+			InitContainerImage:            nil,
+			LivenessProbe:                 nil,
+			NodeSelector:                  nil,
+			PodAffinity:                   nil,
+			PodAntiAffinity:               nil,
+			PriorityClassName:             nil,
+			ReadinessProbe:                nil,
+			SecurityContext:               nil,
+			SideCars:                      nil,
+			SkipChown:                     nil,
+			StartupProbe:                  nil,
+			TerminationGracePeriodSeconds: nil,
+			Tolerations:                   nil,
+			TopologySpreadConstraints:     nil,
+			UpdateStrategy:                nil,
+		}
+
+	case *redpandav1alpha2.PodTemplate:
+		if src.Spec.NodePoolSpec.BrokerTemplate.PodTemplate == nil {
+			return (*redpandav1alpha2.PodTemplate)(nil)
+		}
+
+		return &redpandav1alpha2.PodTemplate{
+			Labels:      src.Spec.NodePoolSpec.BrokerTemplate.PodTemplate.Labels,
+			Annotations: src.Spec.NodePoolSpec.BrokerTemplate.PodTemplate.Annotations,
+			Spec:        (*redpandav1alpha2.PodSpecApplyConfiguration)(src.Spec.NodePoolSpec.BrokerTemplate.PodTemplate.Spec),
+		}
+
+	case redpandav1alpha2.RedpandaStatus:
+		return redpandav1alpha2.RedpandaStatus{
+			Conditions:    src.Status.Conditions,
+			ConfigVersion: src.Status.ConfigVersion,
+			LicenseStatus: (*redpandav1alpha2.RedpandaLicenseStatus)(src.Status.LicenseStatus),
+			NodePools: doMap(src.Status.NodePools, func(src NodePoolStatus) redpandav1alpha2.NodePoolStatus {
+				return redpandav1alpha2.NodePoolStatus{
+					Name:              src.Name,
+					Replicas:          src.Replicas,
+					DesiredReplicas:   src.DesiredReplicas,
+					OutOfDateReplicas: src.OutOfDateReplicas,
+					UpToDateReplicas:  src.UpToDateReplicas,
+					CondemnedReplicas: src.CondemnedReplicas,
+					ReadyReplicas:     src.ReadyReplicas,
+					RunningReplicas:   src.RunningReplicas,
+				}
+			}),
+			// Deprecated fields
```

Comment:
can we just drop these saying all other fields deprecated? or is this for the linter? Also, is that why the `TypeMeta`? For the linter?